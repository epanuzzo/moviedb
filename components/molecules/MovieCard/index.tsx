import React from "react";
import Image from "next/image";
import Link from "next/link";
import YearCircle from "@/components/atoms/YearCircle";

type MenuItemsProps = {
  id: number;
  title: string;
  image: string;
  year: string | number;
};

const MenuItems: React.FC<MenuItemsProps> = ({ id, title, image, year }) => {
  return (
    <Link
      href={`/movie/${id}`}
      className="relative bg-white rounded-md shadow-lg overflow-hidden h-80 hover:cursor-pointer"
    >
      <Image
        src={image}
        alt={title}
        layout="fill"
        objectFit="cover"
        className="absolute inset-0 w-full h-full hover:scale-105 transition-transform"
      />
      <div className="absolute bottom-0 left-0 right-0 p-4 bg-gradient-to-t from-black to-transparent">
        <h3 className="text-lg font-semibold text-white">{title}</h3>
      </div>
      <YearCircle text={year} />
    </Link>
  );
};

export default MenuItems;
