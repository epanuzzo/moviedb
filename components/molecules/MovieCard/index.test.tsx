import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./index";

describe("MovieCard", () => {
  it("renders component", () => {
    const tree = renderer
      .create(
        <MovieCard
          id={1}
          title="Test"
          image="https://placehold.co/600x400"
          rating={5}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders component with alt props", () => {
    const tree = renderer
      .create(
        <MovieCard id={1} title="Test" image="" rating={5} missingImage />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
