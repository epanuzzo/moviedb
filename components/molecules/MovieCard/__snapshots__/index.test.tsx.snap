// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders component 1`] = `
<a
  className="relative bg-white rounded-md shadow-lg overflow-hidden h-80 hover:cursor-pointer"
  href="/movie/1"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <img
    alt="Test"
    className="absolute inset-0 w-full h-full hover:scale-105 transition-transform"
    data-nimg="fill"
    decoding="async"
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    sizes="100vw"
    src="/_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=3840&q=75"
    srcSet="/_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=3840&q=75 3840w"
    style={
      {
        "bottom": 0,
        "color": "transparent",
        "height": "100%",
        "left": 0,
        "objectFit": "cover",
        "objectPosition": undefined,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": "100%",
      }
    }
  />
  <div
    className="absolute bottom-0 left-0 right-0 p-4 bg-gradient-to-t from-black to-transparent"
  >
    <h3
      className="text-lg font-semibold text-white"
    >
      Test
    </h3>
  </div>
  <div
    className="absolute top-4 left-4 bg-white text-black rounded-full flex items-center justify-center w-10 h-10"
  >
    <p
      className="text-sm font-bold"
    >
      5
    </p>
  </div>
</a>
`;

exports[`MovieCard renders component with alt props 1`] = `
<a
  className="relative bg-white rounded-md shadow-lg overflow-hidden h-80 hover:cursor-pointer"
  href="/movie/1"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="w-full h-full bg-gray-800 flex items-center justify-center"
  >
    <p
      className="text-white"
    >
      Image not found
    </p>
  </div>
  <div
    className="absolute bottom-0 left-0 right-0 p-4 bg-gradient-to-t from-black to-transparent"
  >
    <h3
      className="text-lg font-semibold text-white"
    >
      Test
    </h3>
  </div>
  <div
    className="absolute top-4 left-4 bg-white text-black rounded-full flex items-center justify-center w-10 h-10"
  >
    <p
      className="text-sm font-bold"
    >
      5
    </p>
  </div>
</a>
`;
